Document options for pub get and upgrade (#3045)
dart-lang/site-www: src/tools/pub/cmd/pub-get.md

For options that apply to all pub commands, see
[Global options](/tools/pub/cmd#global-options).

> ### `--dry-run` or `-n`

Reviewer: Interesting approach! At first it seemed weird to use h3s (so big, so emphatic), but I like having the options visible in the TOC.

The only badness is that the code font doesn't look great in headings (because of the color maybe? the huge size?) and isn't picked up in the TOC. Still, we could change the code font color/size in h3s to mitigate this.

@parlough, do you have an opinion on making h3s the standard way to document options for tools? Currently we use `<dl>`s instead (e.g. at https://dart.dev/tools/pub/cmd/pub-cache).

The only place I've found where it wouldn't work is in http://localhost:4000/tools/dart2js#options, which has so many options that they're divided into groups. (Weirdly, that page uses h4s instead of h3s, so even the groups don't appear in the TOC.) But I think that exception is fine.

Reviewer 2: @kwalrath I think using h3s seems like a fair idea, for both being present in the TOC but also to simplify contributing and later editing.

Do you have anything in particular in mind in terms of the style change you mentioned? It does seem slightly strange to mix code style with normal text in headers.

Author: I used h3's primarily because I wanted to keep consistent with other pages that listed options ( my basis was https://dart.dev/tools/pub/cmd/pub-lish ).

> With this, pub reports the dependencies that would

Reviewer 2: I'm not sure the full intention here. Reports the dependencies that would _what_?

> With this, pub reports the dependencies that would
> but does not make the changes. This is useful if you

Reviewer: Let's start these with a _-s_ verb. Also, we use [contractions](https://developers.google.com/style/contractions) (especially for _not_) and [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks).

```suggestion
Reports the dependencies that would be made,
but doesn't make the changes.
This option is useful if you
```

want to analyze updates before making them.

> 

Reviewer: Let's add the `--offline` flag to this section. (It can mostly point to previous sections.)

### `--precompile`

> This option makes pub compile executables in its dependencies
> before running the get process.

Reviewer: This doesn't sound right... Wouldn't it compile the executables _after_ getting them?

```suggestion
Creates snapshots of the executables in direct dependencies.
```
